test/test_jit.py

```diff
@@ -1157,6 +1157,47 @@ def forward(self, x):
                        .check("return") \
                        .run(str(get_module_method(m, 'conv', 'conv2d_forward').graph))
 
+    def test_insert_quant_dequant_multi_uses(self):
+        class M(torch.nn.Module):
+            def __init__(self):
+                super(M, self).__init__()
+
+            def forward(self, x, w0, w1, w2):
+                a = F.conv2d(x, w0)
+                b = F.conv2d(a, w1)
+                c = F.conv2d(a, w2)
+                return b + c
+
+        m = torch.jit.script(M())
+        observer = torch.jit.script(Observer())
+        qconfig_dict = {
+            '':
+            QConfig(
+                activation=observer._c,
+                weight=observer._c)
+        }
+        torch._C._jit_pass_insert_observers(m._c, "forward", qconfig_dict, True)
+        data = torch.randn(1, 3, 10, 10, dtype=torch.float)
+        weight = torch.randn(3, 3, 3, 3, dtype=torch.float)
+
+        get_forward(m._c)(data, weight, weight, weight)
+        torch._C._jit_pass_insert_quant_dequant(m._c, "forward", True)
+
+        assert len(m._modules._c.items()) == 0, \
+            'Expected to have zero submodule of conv'
+
+        get_forward(m._c)(data, weight, weight, weight)
+        FileCheck().check("aten::dequantize") \
+                   .check("aten::conv2d") \
+                   .check("aten::quantize_per_tensor") \
+                   .check("aten::dequantize") \
+                   .check_next("aten::dequantize") \
+                   .check("aten::conv2d") \
+                   .check("aten::conv2d") \
+                   .check("return") \
+                   .run(str(get_forward_graph(m._c)))
```

Comment:
I think we want to make sure the dq-conv-q patterns are actually fused as "qconv" as well, you can use `graph_for`(https://github.com/pytorch/pytorch/blob/master/torch/jit/__init__.py#L1632) to get the executed graph I think.

Reply:
I think in this case it's actually good to keep the test as is, without fusion. The test checks that we do support multiple uses and insert q-dq nodes for them - this pass has no business of making sure it gets fused later.

Reply:
OK, as long as we have checks for multiple use that should be OK, but I think we probably want to have integration test for this as well since the end goal is to enable fusion when we have multiple use